### PR TITLE
[14.0][mig]sql_export_excel

### PR DESCRIPTION
**Pull request by hkapatel-initos**
Dependent: sql_request_abstract 
[Fix]: Add header feature in latest v14. #2144

### CODE REVIEW

**Comment by hkapatel-initos** [comment]
@florian-dacosta  It would be great if we could get the reviews here.

**Comment by hkapatel-initos** [comment]
@atchuthan it would be great if we could get the reviews here.

**Comment by hkapatel-initos** [comment]
@dsolanki can you please review this pr?

**Review by fshah-initos — APPROVED** [review]
LGTM

File: sql_export_excel/models/sql_export.py
Comment (florian-dacosta) [inline sql_export_excel/models/sql_export.py]: It is not ok to remove the header parameter here.
If you does it, the header field become useless, but more importante, the whole feature allowing to insert data in an existing xls file (with a lot of formulas for instance) does not work anymore.

This header feature need a modification in `sql_request_abstract` module, that were added here : https://github.com/OCA/server-tools/pull/1659/files

I guess, these modifications are not in version 14, because the port (to v13) was initiated before the PR in v12 was merged.
I believe, it should be added back in this PR to avoid loosing the feature.

Comment (hkapatel-initos) [reply sql_export_excel/models/sql_export.py]: Hello, @florian-dacosta 
As you know the header parameter is removed from the dependent module, 
can I change it in the dependent module, or can you please suggest to me another good way to add a header feature?

Comment (florian-dacosta) [reply sql_export_excel/models/sql_export.py]: You should add it in the dependent module.
It does not really have been removed from dependent module, it is kind on an omission during the migration, so it should be added back. You can do this in this PR IMHO

Comment (hkapatel-initos) [reply sql_export_excel/models/sql_export.py]: suggestions are implemented can you please re-review it

File: sql_export_excel/models/sql_export.py
Comment (florian-dacosta) [inline sql_export_excel/models/sql_export.py]: You can't just copy past the whole method.
The header modification should be done in base module ( `sql_request_abstract`) also because it could be re-used by another module.
Maybe I was not specific enough, I think you can do the header modification in this PR, but it should be done in `sql_request_abstract` directly, the same way it was done in v12, in file  sql_request_abstract/models/sql_request_mixin.py (https://github.com/OCA/server-tools/pull/1659/files).

File: sql_export_excel/tests/test_sql_query_excel.py
Comment (florian-dacosta) [inline sql_export_excel/tests/test_sql_query_excel.py]: I don't see any reason to remove these lines from the test.
If it fails, it means there is a problem with the module

File: sql_export_excel/models/sql_export.py
Comment (hkapatel-initos) [reply sql_export_excel/models/sql_export.py]: The suggestion is implemented can you please re-review it

**Comment by florian-dacosta** [comment]
Thanks a lot, seems good.
Let's wait the other PR to be merged to see if the tests passed but it should be good!

**Comment by florian-dacosta** [comment]
@hkapatel-initos The PR https://github.com/OCA/server-tools/pull/2144 has been merged, could you rebase to make the tests pass?

**Comment by hkapatel-initos** [comment]
> @hkapatel-initos The PR #2144 has been merged, could you rebase to make the tests pass?

Done

**Review by florian-dacosta — APPROVED** [review]

**Comment by hkapatel-initos** [comment]
Hello maintainers,
Can you please try to merge it?

**Review by atchuthan — APPROVED** [review]

**Review by CasVissers-360ERP — APPROVED** [review]

**Pull request merged**
